compiler/runtime/OMRRuntimeAssumptions.cpp

```diff
@@ -40,3 +40,36 @@ void TR::PatchNOPedGuardSite::compensate(bool isSMP, uint8_t *location, uint8_t
    {
    _patchVirtualGuard(location, destination, isSMP);
    }
+
+TR::PatchSites::PatchSites(TR_PersistentMemory *pm, int size) : _size(size), _refCount(0), _patchPoints(NULL)
+   {
+   _patchPoints = new (pm) uint8_t*[size * 2];
+   }
+
+void TR::PatchSites::set(size_t index, uint8_t *location, uint8_t *destination)
+   {
+   TR_ASSERT(index < _size, "Index %d beyond patch sites size %d\n", index, _size);
+   index *= 2;
+   _patchPoints[index] = location;
+   _patchPoints[index + 1] = destination;
+   }
+
+bool TR::PatchSites::equals(PatchSites *other)
```

Comment:
Given that other patch sites compare based on equality of the location should this be a comparison of locations? Is there an ordering constraint on the locations? It would make sense to order them in some cases, but I am unsure of the overheads so would appreciate your thoughts.

Reply:
Ordering has not been enforced in this implementation, as its current uses (in a downstream project) don't provide it cheaply or require it. Considering how large these assumptions can become and how infrequently `equals` is used on `RuntimeAssumptions`, I don't believe its a suitable restriction to make. I have added a `O(n^2)` comparison to `equals` as a tradeoff.